ocs_ci/ocs/resources/pod.py

```diff
@@ -110,14 +110,14 @@ def get_fio_results(self):
             Exception: In case of exception from FIO
         """
         try:
-            if self.fio_thread.running():
-                for sample in TimeoutSampler(
-                    timeout=FIO_TIMEOUT, sleep=3, func=self.fio_thread.done
-                ):
-                    if sample:
-                        return yaml.safe_load(self.fio_thread.result())
-            if self.fio_thread and self.fio_thread.done():
-                return yaml.safe_load(self.fio_thread.result())
+            for sample in TimeoutSampler(
+                timeout=FIO_TIMEOUT, sleep=3, func=self.fio_thread.result
```

Comment:
@RazTamir ,
According to https://docs.python.org/3/library/concurrent.futures.html , for result() function "If timeout is not specified or None, there is no limit to the wait time." so Technically result() never returns in our case unless job is Done.

Is that what expected here ? Correct me if I am missing something.

Reply:
That's true but we have the TimeoutSampler for taking care of it

Reply:
According to my understanding , this line https://github.com/red-hat-storage/ocs-ci/blob/master/ocs_ci/utility/utils.py#L732 will not yield until call is finished.

Am I right ?

Reply:
You probably do.
Let me run some tests and see. 
I will add timeout in case that's true